src/visualization/utilities.jl

```diff
@@ -1143,5 +1251,54 @@ function mesh_plotting_wireframe(pds::PlotDataSeries{<:PlotData2DTriangulated};
   return xyz_wireframe
 end
 
+# Creates a GeometryBasics triangulation for the visualization of a ScalarData2D plot object.
+function global_plotting_triangulation_makie(pd::PlotData2DTriangulated{<:ScalarData};
+                                             set_z_coordinate_zero = false)
+  @unpack x, y, data, t = pd
+
+  makie_triangles = Makie.to_triangles(t)
+
+  # trimesh[i] holds GeometryBasics.Mesh containing plotting information on the ith element.
+  # Note: Float32 is required by GeometryBasics
+  num_plotting_nodes, num_elements = size(x)
+  trimesh = Vector{GeometryBasics.Mesh{3, Float32}}(undef, num_elements)
+  coordinates = zeros(Float32, num_plotting_nodes, 3)
+  for element in Base.OneTo(num_elements)
+    for i in Base.OneTo(num_plotting_nodes)
+      coordinates[i, 1] = x[i, element]
+      coordinates[i, 2] = y[i, element]
+      if set_z_coordinate_zero == false
+        coordinates[i, 3] = data.data[i, element]
+      end
+    end
+    trimesh[element] = GeometryBasics.normal_mesh(Makie.to_vertices(coordinates), makie_triangles)
+  end
+  plotting_mesh = merge([trimesh...]) # merge meshes on each element into one large mesh
+  return plotting_mesh
+end
+
+# Returns a list of `Makie.Point`s which can be used to plot the mesh, or a solution "wireframe"
+# (e.g., a plot of the mesh lines but with the z-coordinate equal to the value of the solution).
+function mesh_plotting_wireframe(pd::PlotData2DTriangulated{<:ScalarData};
```

Comment:
Not sure I understand this method correctly. Why does it return an `xyz_wireframe` while the other `mesh_plotting_wireframe` methods above return tuples of x, y, solution? Are these really methods of the same function or do they just share the same name and are really different things?

Reply:
Good point. They do different things - one generates data for the mesh lines given mesh/solver/cache, while the one specialized to PlotData2DTriangulated converts that data to a GeometryBasics type for Makie. 

Want to rename this version?